esphome/docker-compose.yml

```diff
@@ -0,0 +1,22 @@
+version: "3.7"
+
+services:
+  app_proxy:
+    environment:
+      APP_HOST: $APP_ESPHOME_IP
+      APP_PORT: 6052
+      PROXY_AUTH_ADD: "true"
+
+  server:
+    image: esphome/esphome:2022.6.2@sha256:69d382af56b10006a833eb90d29339b12efdddfb93382764a379c17fe252eda6
+    volumes:
+      - ${APP_DATA_DIR}/data:/config
+      - /etc/localtime:/etc/localtime:ro
+      # Needed for flashing new ESPs
+      - /dev:/dev
+    networks:
+      default:
+        ipv4_address: $APP_ESPHOME_IP
+    restart: on-failure
+    # Needed for flashing new ESPs
+    privileged: true
```

Comment:
Have you tried any cgroup configuration that would grant access to the hardware without 'breaking the sandbox' using `privileged: true` (which essentially whitelists all hardware).

Here is an example where a container accesses a Bitcoin hardware wallet via USB that if I recall correctly works: https://github.com/getumbrel/umbrel/pull/1240/files#diff-c82306ea6e81e5a7b6b58ee6c5dc339fe222eda0936d2c21a22d1d1c5038de5aR24

Reply:
I remember trying it a while ago but it wasn't working properly.

I can give it another try though.

Reply:
Seems to be working

I'll push it in a couple minutes

Reply:
nope, it detects the ESP, but fails to flash
![image](https://user-images.githubusercontent.com/13995143/176267771-6573a58d-8672-4479-a5ba-f4dc975553bc.png)

Reply:
I guess that we only have two options:

1. Keep privileged mode enabled
2. Use the dashboard as is, without the ability to directly flash new ESPs through USB

Reply:
As of right now, we don't want to allow applications to have `privileged: true`. We can have this approved and figure out the flashing behaviour later?

Btw, what did you try in terms of trying to get this to work, as I think it should be possible with similar (ish) cgroup config that I posted above.

Reply:
I tried group `189`, but apparently it's `188` (my bad).

It works now, going to remove `privileged: true` and push.